app/src/pages/playground/InvocationParametersForm.tsx

```diff
@@ -16,113 +30,221 @@ const FormField = ({
       return (
         <Slider
-          label="Top P"
+          label={field.label}
+          isRequired={field.required}
           value={value}
           step={0.1}
-          minValue={0}
-          maxValue={1}
+          minValue={field.minValue}
+          maxValue={field.maxValue}
           onChange={(value) => onChange(value)}
         />
       );
-    case "maxCompletionTokens":
+    case "IntInvocationParameter":
       return (
         <TextField
-          label="Max Completion Tokens"
+          label={field.label}
+          isRequired={field.required}
           value={value?.toString() || ""}
           type="number"
           onChange={(value) => onChange(Number(value))}
         />
       );
-    case "maxTokens":
-      return (
-        <TextField
-          label="Max Tokens"
-          value={value?.toString() || ""}
-          type="number"
-          onChange={(value) => onChange(Number(value))}
-        />
-      );
-    case "stop":
+    case "StringListInvocationParameter":
       if (!Array.isArray(value) && value !== undefined) return null;
       return (
         <TextField
-          label="Stop"
+          label={field.label}
+          isRequired={field.required}
           defaultValue={value?.join(", ") || ""}
           onChange={(value) => onChange(value.split(/, */g))}
         />
       );
-    case "seed":
+    case "StringInvocationParameter":
       return (
         <TextField
-          label="Seed"
+          label={field.label}
+          isRequired={field.required}
           value={value?.toString() || ""}
-          type="number"
-          onChange={(value) => onChange(Number(value))}
+          type="text"
+          onChange={(value) => onChange(value)}
         />
       );
+    case "BooleanInvocationParameter":
+      // TODO: add checkbox
+      return null;
     default:
       return null;
   }
 };
 
-export type InvocationParametersChangeHandler = <
-  T extends keyof ModelConfig["invocationParameters"],
->(
-  parameter: T,
-  value: ModelConfig["invocationParameters"][T]
-) => void;
+const toCamelCase = (str: string) =>
+  str.replace(/_([a-z])/g, (_, char) => char.toUpperCase());
+
+const getInvocationParameterValue = (
+  field: InvocationParameter,
+  parameterInput: InvocationParameterInput
+): string | number | string[] | boolean | null | undefined => {
+  if (field.invocationInputField === undefined) {
+    throw new Error("Invocation input field is required");
+  }
+  return parameterInput[
+    toCamelCase(field.invocationInputField) as keyof InvocationParameterInput
+  ];
+};
+
+const makeInvocationParameterInput = (
+  field: InvocationParameter,
+  value: string | number | string[] | boolean | undefined
+): InvocationParameterInput | null => {
+  if (field.invocationName === undefined) {
+    throw new Error("Invocation name is required");
+  }
+  if (field.invocationInputField === undefined) {
+    throw new Error("Invocation input field is required");
+  }
```

Comment:
when would these things happen, shouldn't these always be defined coming from the api?

Reply:
not sure what this means

Reply:
when would field.invocationName or field.invocationInputField be undefined?

Reply:
The generated TS types from relay claims every field value is optional when it generates types.

I think it is because an InvocationParameter can be one of many union types, so it just mashes all the fields together and marks them all as optional.

Reply:
<img width="616" alt="image" src="https://github.com/user-attachments/assets/221e8423-26d1-44a7-a5d1-78720df5a40f">